spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/InitCommand.java

```diff
@@ -128,16 +129,20 @@ protected void options() {
 			otherOptions();
 		}
 
+		/**
+		 * Supports both kebab-case and camelCase as project CLI Options. camelCase to be
+		 * deprecated as part of future releases
+		 */
 		private void projectGenerationOptions() {
-			this.groupId = option(Arrays.asList("groupId", "g"), "Project coordinates (for example 'org.test')")
-					.withRequiredArg();
-			this.artifactId = option(Arrays.asList("artifactId", "a"),
+			this.groupId = option(Arrays.asList("groupId", "group-id", "g"),
+					"Project coordinates (for example 'org.test')").withRequiredArg();
+			this.artifactId = option(Arrays.asList("artifactId", "artifact-id", "a"),
 					"Project coordinates; infer archive name (for example 'test')").withRequiredArg();
 			this.version = option(Arrays.asList("version", "v"), "Project version (for example '0.0.1-SNAPSHOT')")
 					.withRequiredArg();
 			this.name = option(Arrays.asList("name", "n"), "Project name; infer application name").withRequiredArg();
 			this.description = option("description", "Project description").withRequiredArg();
-			this.packageName = option("package-name", "Package name").withRequiredArg();
+			this.packageName = option(Arrays.asList("packageName", "package-name"), "Package name").withRequiredArg();
```

Comment:
The plan is to use kebab case consistently, not introducing a mixed case for new properties.

Reply:
I thought we were going to support both camelCase and kebab-case until we've improved the output of `spring init --list` so that it converts the server's output from the camelCase that it expects to the kebab-case that the CLI prefers. That improvement to `spring init --list` was to be handled separately.

Reply:
Ah ok, right. I understand now. Yes, so we should have to consistent indeed but we need to deprecate them.